### ISSUE TITLE
Impact of #177 on Captions and Labels

### ISSUE THREAD

**Issue by cdhutch** [fletcher/MultiMarkdown-6#194]
Prior to the incorporation of #177 if I made a table such as :

|     Name    |         Role        |
|-------------|---------------------|
| Arthur      | King of All Britons |
| Sir Robin   | Chicken             |
[Knights of the Round Table][knights]

the resulting LaTeX would include the following caption and label lines:
>\caption{Knights of the Round Table}
>\label{knights}

Since #177, the code becomes:
>\caption[knights]{Knights of the Round Table}
>\label{knights}

The effect of this change seems to be that the table's short caption and label are now linked. This requires me to either change my internal table/graphics labels to be "prettier" show they show up nicely in the ToC.

Does the current MMD code offer syntax that allows the user to specify a different value for the short caption field than from the label field or to forego the short caption capability?

Thanks for the great software development.

**Comment by fletcher** [comment]
> Does the current MMD code offer syntax that allows the user to specify a different value for the short caption field than from the label field or to forego the short caption capability?

No.  I agreed with @MArpogaus that the full caption should not be used in the ToC for images and tables, which should include more of a description of the image/table than the full caption.

My first suggestion would be to use a "prettier" short caption/label.  This can still be succinct and memorable, but provides a cleaner list of figures and still enables cross-reference linking.

Alternatively, you can always modify MultiMarkdown to work differently in the manner you prefer (the benefit of open source code....)

(Presumably you could also redefine the `\caption[]{}` LaTeX command to ignore the first argument, which would effectively revert the behavior to the old approach of using only the long caption, but I have not tried this myself.)

**Comment by cdhutch** [comment]
Thanks for the response. Your first suggestion would work well going forward, but I have so many documents that are "pre #177" that it would be a lot of work to go back and change them. Modifying MMD code itself would provide a low ROI, given that I haven't touched C since 1991. Instead, I'll pursue redefining `\caption[]{}` in my MMD latex config files.

I'll report back with that solution once I get it working.
